### PR TITLE
Modify teacher service

### PR DESCRIPTION
**Pull request by gauriramesh**

### CODE REVIEW

**Comment by hassanhabib** [comment]
This PR is not test driven.
It doesn't have pure commits, no track history whether the test was written first or after the implementation
There's a 3.5 hours video that explains how to commit, test drive and implement modifying an entity in here: https://www.youtube.com/watch?v=FLsHIDe3cNs

File: OtripleS.Web.Api/Models/Teachers/Exceptions/NullTeacherException.cs [hunk 1]
```diff
@@ -0,0 +1,9 @@
+﻿using System;
+
+namespace OtripleS.Web.Api.Models.Teachers.Exceptions
+{
+    public class NullTeacherException : Exception
+    {
+        public NullTeacherException() : base($"The teacher is null.") { }
```
Comment (hassanhabib) [inline OtripleS.Web.Api/Models/Teachers/Exceptions/NullTeacherException.cs:7]: Why do we need string interpolation here if you are not passing any parameters.

File: OtripleS.Web.Api/Services/Teachers/TeacherService.Validations.cs [hunk 2]
```diff
@@ -28,5 +28,129 @@ private void ValidateStorageTeacher(Teacher maybeTeacher, Guid teacherId)
...
+
+        private void ValidateTeacherOnModify(Teacher teacher)
+        {
+            ValidateTeacher(teacher);
+            ValidateTeacherId(teacher.Id);
+            ValidateTeacherIds(teacher);
+            ValidateTeacherStrings(teacher);
+            ValidateTeacherDates(teacher);
+            ValidateDatesAreNotSame(teacher);
+            ValidateUpdatedDateIsRecent(teacher);
```
Comment (hassanhabib) [inline OtripleS.Web.Api/Services/Teachers/TeacherService.Validations.cs:40]: Where are the tests for any of these?

File: OtripleS.Web.Api/Services/Teachers/TeacherService.Validations.cs [hunk 3]
```diff
@@ -28,5 +28,129 @@ private void ValidateStorageTeacher(Teacher maybeTeacher, Guid teacherId)
...
+                throw new InvalidTeacherInputException(
+                    parameterName: nameof(Teacher.UpdatedDate),
+                    parameterValue: teacher.UpdatedDate);
+            }
+        }
+
+        private static bool IsInvalid(string input) => String.IsNullOrWhiteSpace(input);
+        private static bool IsInvalid(Guid input) => input == default;
+        private bool IsDateNotRecent(DateTimeOffset dateTime)
+        {
```
Comment (hassanhabib) [inline OtripleS.Web.Api/Services/Teachers/TeacherService.Validations.cs:148]: Separate multi-liner methods from single-liner methods with a new empty line please

**Comment by viralpandya** [comment]
Gauri, Please look into this https://github.com/hassanhabib/OtripleS/pull/42 for your reference.

**Comment by hassanhabib** [comment]
Closing this PR in favor of https://github.com/hassanhabib/OtripleS/pull/42

**Comment by viralpandya** [comment]
Hi Gauri, 

I am taking this branch and working on modify. Please let me know if you have any questions.

Regards,
Viral

**Pull request closed without merging**